docs: tw697-das-rpc-method-ref (#3213)
OffchainLabs/arbitrum-docs: docs/launch-arbitrum-chain/02-configure-your-chain/common/data-availability/data-availability-committees/das-rpc-method-reference.mdx

---
title: 'DAS RPC method reference'
sidebar_label: 'DAS RPC reference'
description: 'Discover and learn about the DAS RPC methods.'
author: pete-vielhaber
sme: Jason-W123
user_story: As a current Arbitrum chain owner, I want to refer to the DAS RPC methods.
content_type: reference
---

> This document covers the JSON-RPC methods exposed by Arbitrum Data Availability Servers (DAS) and the newer DAProvider interface, as defined in the [Arbitrum Chain SDK](https://github.com/OffchainLabs/arbitrum-chain-sdk/) node configuration types [`NodeConfig.generated.ts`](https://github.com/OffchainLabs/arbitrum-chain-sdk/blob/main/src/types/NodeConfig.ts)

Reviewer: ```suggestion
This document covers the JSON-RPC methods exposed by Arbitrum Data Availability Servers (DAS) and the newer DAProvider interface, as defined in the [Arbitrum Chain SDK](https://github.com/OffchainLabs/arbitrum-chain-sdk/) node configuration types [`NodeConfig.generated.ts`](https://github.com/OffchainLabs/arbitrum-chain-sdk/blob/main/src/types/NodeConfig.generated.ts)
```

| Direction           | Client (batch poster) -> DAS server                                                         |

**Behavior:**

- Called once at the beginning of a chunked store operation
- Returns a stream/session identifier used by subsequent `das_sendChunk` calls
- The server allocations resources to receive the incoming data stream

**When invoked:**

- The batch poster calls this method on each DAS backend in the RPC aggregator's backend list
- Only used when `enable-chunked-store` is `true` (default)

### `das_sendChunk`

Sends a single chunk of batch data within an active chunked store session.

| Field               | Value                                                                                       |
| ------------------- | ------------------------------------------------------------------------------------------- |
| Default method name | `das_sendChunk`                                                                             |
| Config path         | `node.data-availability.rpc-aggregator.das-rpc-client.data-stream.rpc-methods.stream-chunk` |

> | Config key          | `start-chunk`                                                                               |

Reviewer: ```suggestion
| Config key          | `stream-chunk`                                                                               |
```

### `daprovider_commitChunkedStore`

Finalizes a chunked store session on the DA provider server.

| Field               | Value                                                      |
| ------------------- | ---------------------------------------------------------- |
| Default method name | `daprovider_commitChunkedStore`                            |
| Config path         | `node.da-provider.data-stream.rpc-methods.finalize-stream` |
| Config key          | `finalize-stream`                                          |

Functionally equivalent to `das_commitChunkedStore`.

### `daprovider_store`

Single-shot store method—sends the entire payload in one RPC call. Used when data streaming is disabled. (`use-data-streaming: false`).

| Field               | Value                               |
| ------------------- | ----------------------------------- |
| Default method name | `daprovider_store`                  |
| Config path         | `node.da-provider.store-rpc-method` |
| Config key          | `store-rpc-method`                  |

**Behavior:**

- Sends the complete batch data in a single request
- Simpler than a chunked store but limted by HTTP body size constraints

> - Disabled by default in favor of the streaming protocol

Reviewer: ```suggestion
- Enabled by default. Set `use-data-streaming: true` to switch to the chunked protocol.
```
